arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -1233,7 +1237,11 @@ bool HeartbeatThread::handlePlanChangeCoordinator(uint64_t currentPlanVersion) {
 
       if (r == ids.end()) {
         // local database not found in the plan...
-        databaseFeature.dropDatabase(id, true);
+        TRI_vocbase_t* db = databaseFeature.useDatabase(id);
+        TRI_ASSERT(db);
+        std::string dbName = db ? db->name() : "n/a";
```

Comment:
Wondering if this is safe...
First of all, could someone else have dropped the database before the `useDatabase(id)`, so `useDatabase(id)` returns a nullptr? If `db` is never a nullptr, then it is strange that we check it in the expression `db ? db->name() : "n/a"`.
And finally, every successful `useDatabase(...)` should be accompanied by a call to `db->release()`, otherwise the database object will never get deleted.

Reply:
Ah, yes, forgot about the release. I assume that no one else will drop the database before the use, but just in case, I added the null check. Figured go ahead and crash out with the assert in maintainer mode, but be extra safe in production.